File: 240305/개발팀의 능력/the-capabilities-of-the-development-team.py
import sys

num=list(map(int,input().split()))

def get_diff(i,j,k,l):
    sum1=num[i]+num[j]
    sum2=num[k]+num[l]
    sum3=sum(num)-sum1-sum2
    if sum1==sum2==sum3:
        return sys.maxsize
    return max([sum1,sum2,sum3])-min([sum1,sum2,sum3])

ans=sys.maxsize
flag=0
for i in range(5):
    for j in range(i+1,5):
        for k in range(5):
            for l in range(k+1,5):
                if i==j or i==k or j==k or j==l:
                    continue
                if num[i]==num[j] or num[k]==num[l]:

                    continue
                ans=min(ans,get_diff(i,j,k,l))
                #print(i,j,k,l,get_diff(i,j,k,l))
                flag=1
if flag==0:
    print(-1) 
else: 
    print(ans)

File: 260113/개발팀의 능력/the-capabilities-of-the-development-team.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split("\n");

const arr = input[0].split(" ").map(Number);

// Please Write your code here.
let sum = 0;
arr.forEach((num)=>{
    sum+=num;
})

let ans = 5000;
for(let i=0; i<5; i++){
    for(let j=i+1; j<5; j++){
        for(let k=0; k<5; k++){
            if(k===i|| k===j) continue;
            let t1=arr[i]+arr[j];
            let t2 = arr[k];
            let t3 = sum-(arr[i]+arr[j]+arr[k]);
            if(t1===t2 || t2===t3 || t3===t1) continue;
            ans = Math.min(ans, Math.max(t1,t2,t3)-Math.min(t1,t2,t3));
        }
    }
}

console.log(ans===5000?-1:ans);